sdk/dnsresolver/Microsoft.Azure.Management.DnsResolver/tests/ScenarioTests/DnsResolverTests.cs

```diff
@@ -0,0 +1,643 @@
+            Action createDnsResolverAction = () => this.DnsResolverManagementClient.DnsResolvers.CreateOrUpdate(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: existingDnsResolver.Name,
+                parameters: dnsResolverForCreation);
+
+            createDnsResolverAction.Should().NotBeNull();
+            createDnsResolverAction.Should().Throw<CloudException>().Which.Response.StatusCode.Should().Be(HttpStatusCode.Conflict);
+        }
+ 
+        [Fact]
+        public void PutDnsResolver_ResolverNotExistsIfNoneMatchSuccess_ExpectResolvereCreated()
+        {
+            var resourceGroupName = this.CreateResourceGroup().Name;
+            var dnsResolverName = TestDataGenerator.GenerateDnsResolverName();
+            var dnsResolverForCreation = TestDataGenerator.GenerateDnsResolver(location: Constants.DnsResolverLocation,  subscriptionId: this.SubscriptionId, resourceGroupName: resourceGroupName);
+            var ifNoneMatch = "*";
+
+            var createdDnsResolver = this.DnsResolverManagementClient.DnsResolvers.CreateOrUpdate(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: dnsResolverName,
+                parameters: dnsResolverForCreation,
+                ifNoneMatch: ifNoneMatch);
+
+            createdDnsResolver.Should().NotBeNull();
+            createdDnsResolver.ProvisioningState.Should().BeEquivalentTo(Constants.ProvisioningStateSucceeded.ToString());
+        }
+
+        [Fact]
+        public void GetDnsResolver_ResolverExistsByResolverName_ExpectResolverRetrieved() 
+        {
+            var resourceGroupName = this.CreateResourceGroup().Name;
+            var createdDnsResolver = this.CreateDnsResolver(resourceGroupName: resourceGroupName);
+
+            var retrievedDnsResolver = this.DnsResolverManagementClient.DnsResolvers.Get(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: createdDnsResolver.Name);
+
+            retrievedDnsResolver.Should().NotBeNull();
+            retrievedDnsResolver.ProvisioningState.Should().Be(Constants.ProvisioningStateSucceeded);
+        }
+
+        [Fact]
+        public void GetDnsResolver_ResolverNotExistsByResolverName_ExpectFailure()
+        {
+            var resourceGroupName = this.CreateResourceGroup().Name;
+            var dnsResolverName = TestDataGenerator.GenerateDnsResolverName();
+
+            Action retrieveDnsResolverAction = () => this.DnsResolverManagementClient.DnsResolvers.Get(
+                resourceGroupName: resourceGroupName,
+                dnsResolverName: dnsResolverName);
+
+            retrieveDnsResolverAction.Should().NotBeNull();
+            retrieveDnsResolverAction.Should().Throw<CloudException>().Which.Response.StatusCode.Should().Be(HttpStatusCode.NotFound);
+        }
+
+        [Fact]
+        public void ListDnsResolverByVirtualNetwork_ValidVirtualNetworkAttachedToResolver_ExpectResolverRetrieved()
```

Comment:
At least one of these List cases should confirm the listed resolvers are what we created given that the List flow invokes different repository APIs.

Reply:
Have method `ValidateListedResolverIsExpected` now to do deep comparisons.